skrf/mathFunctions.py

```diff
@@ -1258,3 +1258,58 @@ def is_positive_semidefinite(mat: npy.ndarray, tol: float = ALMOST_ZERO) -> bool
     except npy.linalg.LinAlgError:
         return False
     return npy.all(v > -tol)
+
+def rsolve(A: npy.ndarray, B: npy.ndarray) -> npy.ndarray:
+    r"""Solves x @ A = B.
+
+    Calls numpy.linalg.solve with transposed matrices.
+
+    Input should have dimension of similar to (nfreqs, nports, nports).
+
+    Parameters
+    ----------
+    A : npy.ndarray
+    B : npy.ndarray
+
+    Returns
+    -------
+    x : npy.ndarray
+    """
+    return npy.transpose(npy.linalg.solve(npy.transpose(A, (0, 2, 1)).conj(),
+            npy.transpose(B, (0, 2, 1)).conj()), (0, 2, 1)).conj()
+
+def nudge_eig(mat: npy.ndarray, cond: float = 1e-9, min_eig: float = 1e-12) -> npy.ndarray:
```

Comment:
nice. Did you use a reference for that? I'd like to learn more on that.

Reply:
I don't have any specific references for this method. I first tried adding a small constant times diagonal matrix if the condition number of the matrix was very big but that could fail in some cases. Since condition number is calculated from eigenvalues it makes sense to just change the problematic eigenvalues directly to improve the condition number.

Reply:
OK.